### PR TITLE
Prioritise specified transaction gas to allow gasless network transactions

### LINKED ISSUE #3305: Metamask mobile requires gas fees even for gasless, private networks
**Describe the bug**
Currently, we are using Metamask Mobile, on both **iOS & Android**, connected to a custom, private network in QBS _Quorum Blockchain Services_. This network has no currency at all, and is unable to generate any.

However, when interacting with our deployed smart contracts on the network, Metamask Mobile still requests for the minimum gas fees. All our smart contracts and interactions are set at **0 fees**. Attached below are some screenshots of the issue.

This issue **DOES NOT** exist on the Metamask Browser Extension.

**Screenshots**
This image shows that the interactions are set to **0**

code-snippet

The error that occurs on the Metamask **iOS & Android** versions of the app that shows up. Note that we are unable to transact when the wallet has no currency.

MetamaskMobile_Transaction_error

**To Reproduce**
_Steps to reproduce the behavior_
1. Deploy a smart contract on the Consensys QBS network
2. Try to interact with contract with a newly created Metamask account on Metamask Mobile

**Expected behavior**
The transaction should be allowed to go through when on a gasless network, with an account with no currency, instead of blocking the transaction.

We have also done extra testing on a network where gas prices are **0**, but are able to secure currency. Even though the popup still comes up saying that we require gas fees, after the transaction is processed **no gas fees are consumed**, which confirms that it is not an issue with our code/smart contracts.

This issue **does not** exist on the metamask browser extension. But we are targetting users on the Metamask mobile application.

**Smartphone (please complete the following information):**
 - Device: iPhone 8/11/12 Pro as well as Samsung Android
 - App Version: v3.4.1

-------------------------------------------------------------
_to be added after bug submission by internal support / PM_
**Severity**

- How critical is the impact of this bug on a user?
- Add stats if available on % of customers impacted
- Is this visible to all users?
- Is this tech debt?

### PR DESCRIPTION
**Pull request by Fatxx**
**Description**

On extension the gas fee specified in the transaction is prioritise over the average gas price.
This PR implements that logic and further enable gasless transactions on networks like ConsenSys Quorum.

**How to reproduce**

- I followed https://consensys.net/docs/goquorum/en/latest/tutorials/quorum-dev-quickstart/ to run Quorum locally
- Run the Pet Shop Daap
- Try to adopt a pet

**Checklist**

* [x] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

**Issue**

Resolves #3305

### CODE REVIEW

File: app/components/UI/TransactionEditor/index.js [hunk 1]
```diff
@@ -523,14 +523,14 @@ class TransactionEditor extends PureComponent {
...
-			transaction: { value, from, assetType },
+			transaction: { value, from, assetType, gasPrice },
 		} = this.props;
 
 		const checksummedFrom = safeToChecksumAddress(from) || '';
 		const fromAccount = this.props.accounts[checksummedFrom];
 		const { balance } = fromAccount;
 		const weiBalance = hexToBN(balance);
-		const totalGasValue = hexToBN(totalGas);
+		const totalGasValue = gasPrice ? hexToBN(gasPrice) : hexToBN(totalGas);
```
Comment (andreahaku) [inline app/components/UI/TransactionEditor/index.js:533]: what about:
`const totalGasValue = hexToBN(gasPrice || totalGas);`

**Review by andreahaku — APPROVED** [review]

File: app/components/UI/TransactionEditor/index.js [hunk 2]
```diff
@@ -523,14 +523,14 @@ class TransactionEditor extends PureComponent {
...
-			transaction: { value, from, assetType },
+			transaction: { value, from, assetType, gasPrice },
 		} = this.props;
 
 		const checksummedFrom = safeToChecksumAddress(from) || '';
 		const fromAccount = this.props.accounts[checksummedFrom];
 		const { balance } = fromAccount;
 		const weiBalance = hexToBN(balance);
-		const totalGasValue = hexToBN(totalGas);
+		const totalGasValue = hexToBN(gasPrice || totalGas);
```
Comment (andrepimenta) [inline app/components/UI/TransactionEditor/index.js:533]: The gasPrice should not be get prioritized when compared to totalGas. totalGas comes from our API that is much better at predicting prices than what comes from gasPrice. But, for custom networks where our API does not work, we should already be using the gasPrice. So this shouldn't be necessary, there is a bug somewhere on our gas controller probably.

**Review by andrepimenta — CHANGES_REQUESTED** [review]

**Review by andrepimenta — APPROVED** [review]
LGTM!

**Review by cortisiko — APPROVED** [review]
I synced with @gantunesr  and @Fatxx  on this. We are 🌮

**Pull request merged**